src/hotspot/share/prims/unsafe.cpp

```diff
@@ -783,21 +783,23 @@ UNSAFE_ENTRY(void, Unsafe_Park(JNIEnv *env, jobject unsafe, jboolean isAbsolute,
 
 UNSAFE_ENTRY(void, Unsafe_Unpark(JNIEnv *env, jobject unsafe, jobject jthread)) {
   if (jthread != nullptr) {
-    ThreadsListHandle tlh;
-    JavaThread* thr = nullptr;
-    oop java_thread = nullptr;
-    (void) tlh.cv_internal_thread_to_JavaThread(jthread, &thr, &java_thread);
-    if (java_thread != nullptr) {
-      // This is a valid oop.
-      if (thr != nullptr) {
-        // The JavaThread is alive.
-        Parker* p = thr->parker();
-        HOTSPOT_THREAD_UNPARK((uintptr_t) p);
-        p->unpark();
+    oop thread_oop = JNIHandles::resolve_non_null(jthread);
+    if (java_lang_Thread::thread(thread_oop) != nullptr) {
```

Comment:
Suggestion for a comment after this line:
```
// We initially have a live JavaThread, so try to capture it in a ThreadsListHandle
```

Reply:
I've added:
```
    if (java_lang_Thread::thread(thread_oop) != nullptr) {
      // Try to capture the live JavaThread in a ThreadsListHandle:
      ThreadsListHandle tlh; // Provides memory barrier
```